constrain-lib/src/main/resources/schema/dependency-constraints-schema.json

```diff
@@ -0,0 +1,134 @@
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "type": "object",
+  "title": "Dependency Constraints Schema.",
+  "description": "",
+  "required": [
+    "version",
+    "dependencyConstraints"
+  ],
+  "properties": {
+    "version": {
+      "type": "string",
+      "title": "Version",
+      "description": "Version of the schema.",
+      "default": "1.0.0",
+      "examples": [
+        "1.0.0"
+      ]
+    },
+    "dependencyConstraints": {
+      "$ref": "#/$defs/dependencyConstraints"
+    }
+  },
+  "additionalProperties": false,
+  "$defs": {
+    "dependencyConstraints": {
+      "type": "array",
+      "title": "Constraints",
+      "description": "A collection of Dependency Constraints.",
+      "items": {
+        "$ref": "#/$defs/dependencyConstraint"
+      }
+    },
+    "dependencyConstraint": {
+      "type": "object",
+      "title": "Constraint",
+      "description": "A single Dependency Constraint.",
+      "required": [
+        "groupId",
+        "name",
+        "suggestedVersion",
+        "because"
+      ],
+      "properties": {
+        "groupId": {
+          "type": "string",
+          "title": "Group",
+          "description": "The group of the dependency.",
+          "examples": [
+            "com.example"
+          ]
+        },
+        "name": {
+          "type": "string",
+          "title": "Name",
+          "description": "The name of the dependency.",
+          "examples": [
+            "example-dependency"
+          ]
+        },
+        "suggestedVersion": {
+          "type": "string",
+          "title": "Suggested Version",
+          "description": "The suggested version of the dependency.",
+          "examples": [
+            "1.0.0"
+          ]
+        },
+        "rejectedVersions": {
+          "type": "array",
+          "title": "Rejected Versions",
+          "description": "The list of versions that are rejected by this constraint.",
+          "items": {
+            "type": "string",
+            "title": "Rejected Version",
+            "description": "This supports either specific versions or the Gradle version range format: https://docs.gradle.org/current/userguide/single_versions.html",
+            "examples": [
+              "1.0.0",
+              "[1.0,1.4]",
+              "[1.0,1.4)"
```

Comment:
Beware that if the interval is half-opened or opened and no suggestedVersion is available (if optional as suggested in a previous comment), this could lead to an unresolvable constraint.

Reply:
Yea, this is exactly why I'm thinking that requiring a `suggestedVersion` is the right thing to do here.

Reply:
@JLLeitschuh I think you are right. I was thinking about both approach, but if suggestedVersion is mandatory, then this concern is no longer valid.